nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -21,60 +21,143 @@
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
 import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Level;
+import org.apache.log4j.spi.LoggingEvent;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mortbay.jetty.Server;
+import org.mortbay.jetty.handler.AbstractHandler;
 import org.nuxeo.common.Environment;
+import org.nuxeo.common.utils.ZipUtils;
 import org.nuxeo.connect.NuxeoConnectClient;
+import org.nuxeo.connect.connector.http.ConnectUrlConfig;
 import org.nuxeo.connect.update.PackageState;
 import org.nuxeo.launcher.config.TomcatConfigurator;
 import org.nuxeo.launcher.connect.fake.LocalConnectFakeConnector;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.Jetty;
+import org.nuxeo.runtime.test.runner.JettyFeature;
+import org.nuxeo.runtime.test.runner.LogCaptureFeature;
 
 /**
  * @since 8.4
  */
+@RunWith(FeaturesRunner.class)
+@Features({ LogCaptureFeature.class, JettyFeature.class })
+@Jetty(port = ConnectUrlConfig.CONNECT_TEST_MODE_PORT)
 public class TestConnectBroker {
 
+    final String TEST_STORE_PATH = "src/test/resources/packages/store";
+
+    final String TEST_LOCAL_ONLY_PATH = TEST_STORE_PATH + "/local-only";
+
+    final File testStore = new File(TEST_STORE_PATH);
+
+    final File nuxeoHome = new File("target/launcher");
+
+    @Inject
+    Server server;
+
+    @Inject
+    LogCaptureFeature.Result logCaptureResult;
+
+    public static class PkgRequestLogFilter implements LogCaptureFeature.Filter {
+        @Override
+        public boolean accept(LoggingEvent event) {
+            return event.getLevel().isGreaterOrEqual(Level.INFO) && (event.getLoggerName().contains("ConnectBroker")
+                    || event.getLoggerName().contains("PackagePersistence")
+                    || event.getLoggerName().contains("LocalDownloadingPackage"));
+        }
+    }
+
+    public class FakeConnectDownloadHandler extends AbstractHandler {
+
+        @Override
+        public void handle(String target, HttpServletRequest request, HttpServletResponse response, int dispatch)
+                throws IOException, ServletException {
+            if (target.startsWith("/test")) {
+                String pkgId = target.substring(target.lastIndexOf("/") + 1);
+                File pkgZip = new File(testStore, pkgId + ".zip");
+                if (pkgZip.exists()) {
+                    response.setContentLength((int) pkgZip.length());
+                    try (ServletOutputStream os = response.getOutputStream()) {
+                        byte[] byteArray = FileUtils.readFileToByteArray(pkgZip);
```

Comment:
Even if it's a test, don't read things to an in-memory bytes array. IOUtils has methods to copy from file/inputstream to outputstream.